tests/integration/team_test.py

```diff
@@ -17,34 +17,106 @@
 from libnmstate.schema import Team
 
+from .testlib import assertlib
+from .testlib import cmd as libcmd
+
+
+RPM_NM_TEAM_PLUGIN_PATH = ("rpm", "-ql", "NetworkManager-team")
+
+SYSTEMCTL_RESTART_NM_CMD = ("systemctl", "restart", "NetworkManager")
 
 TEAM0 = "team0"
 
-TEAM0_STATE = {
-    Interface.KEY: [
-        {
-            Interface.NAME: TEAM0,
-            Interface.TYPE: InterfaceType.TEAM,
-            Team.CONFIG_SUBTREE: {
-                Team.PORT_SUBTREE: [{Team.Port.NAME: "eth1"}]
+
+@pytest.mark.skipif(
+    os.environ.get("CI") == "true",
+    reason="Team kmod not available in Travis CI",
+)
+def test_create_team_iface():
+    with team_interface(TEAM0) as team_state:
+        assertlib.assert_state(team_state)
+
+
+@pytest.mark.skipif(
+    os.environ.get("CI") == "true",
+    reason="Team kmod not available in Travis CI",
+)
+def test_edit_team_iface():
+    with team_interface(TEAM0) as team_state:
+        team_state[Team.CONFIG_SUBTREE] = {
+            Team.RUNNER_SUBTREE: {
+                Team.Runner.NAME: Team.Runner.RunnerMode.LOAD_BALANCE
             },
+            Team.PORT_SUBTREE: [{Team.Port.NAME: "eth1"}],
         }
-    ]
-}
+        libnmstate.apply(team_state)
 
 
-@pytest.mark.xfail(
-    raises=NmstateNotImplementedError,
-    reason="Team interface is not supported yet",
-    strict=True,
-)
-def test_sriov_not_implemented():
-    libnmstate.apply(TEAM0_STATE)
+def test_nm_team_plugin_missing():
+    with disable_nm_team_plugin():
+        with pytest.raises(NmstateLibnmError):
+            libnmstate.apply(
+                {
+                    Interface.KEY: [
+                        {
+                            Interface.NAME: TEAM0,
+                            Interface.TYPE: InterfaceType.TEAM,
+                            Interface.STATE: InterfaceState.UP,
+                        }
+                    ]
+                }
+            )
+
+
+@contextmanager
+def team_interface(ifname):
+    desired_state = {
+        Interface.KEY: [
+            {
+                Interface.NAME: ifname,
+                Interface.TYPE: InterfaceType.TEAM,
+                Interface.STATE: InterfaceState.UP,
+            }
+        ]
+    }
+    libnmstate.apply(desired_state)
+    try:
+        yield desired_state
+    finally:
+        team_dstate = desired_state[Interface.KEY][0]
+        team_dstate[Interface.STATE] = InterfaceState.ABSENT
+        libnmstate.apply(desired_state)
+
+
+@contextmanager
+def disable_nm_team_plugin():
+    rpm_team_plugin = libcmd.exec_cmd(RPM_NM_TEAM_PLUGIN_PATH)[1]
+    plugin_path = re.search(r".*so", rpm_team_plugin).group()
+    disable_team_plugin = (
+        "mv",
+        f"{plugin_path}",
+        "/tmp/",
+    )
+    libcmd.exec_cmd(disable_team_plugin)
+    libcmd.exec_cmd(SYSTEMCTL_RESTART_NM_CMD)
+    try:
+        yield
+    finally:
+        ls_tmp = libcmd.exec_cmd(("ls", "/tmp/"))[1]
+        plugin_filename = re.search(r".*so", ls_tmp).group()
```

Comment:
Actually, just use `glob.glob("/tmp/*.so")`, no need for listdir.

Reply:
Actually it would be better to copy the plugin to a temp file created with `tempfile.NamedTemporaryFile()` so that you do not need to do any globing that might fail on non CI systems.

Reply:
Not needed anymore.